bin/varnishd/mgt/mgt_jail_linux.c

```diff
@@ -49,10 +49,60 @@
 #include "common/heritage.h"
 
 static int
-vjl_init(char **args)
+vjl_set_thp(const char *arg)
 {
+	if (!strcmp(arg, "enable")) {
+		if (prctl(PR_SET_THP_DISABLE, 0, 0, 0, 0) != 0) {
+			MGT_Complain(C_INFO,
+			    "Could not enable Transparent Hugepages: %s (%d)",
+			    VAS_errtxt(errno), errno);
+		}
+	} else if (!strcmp(arg, "disable") ||
+		    !strcmp(arg, "default")) {
+		if (prctl(PR_SET_THP_DISABLE, 1, 0, 0, 0) != 0) {
+			MGT_Complain(C_INFO,
+			    "Could not disable Transparent Hugepages: %s (%d)",
+			    VAS_errtxt(errno), errno);
+		}
+	} else {
```

Comment:
Please add a option to do nothing a.k.a NOP. I guess the "default" option could be that.
Make sure to print which selection has been selected, maybe also retrieve what option the kernel use?
You could use `PR_GET_THP_DISABLE` for that.

Reply:
It's actually "ignore" that will do nothing. The "default" one was the result of a misunderstanding and it will go away.